### ISSUE TITLE
Returning from "Go to program" fails, ends up in Videos

### ISSUE THREAD

**Issue by dagwieers** [add-ons/plugin.video.vrt.nu#435]
### Describe the bug ###
When going back after selecting "Go to program" from the context menu, Kodi doesn't go back to the recent/offline listing but ends up in Kodi's Videos window.

This appears to be a bug in Kodi rather than an add-on issue.

We use the `ActivateWindow(window,dir,return)` call which should bring you back to the originating listing when you go back.

### To Reproduce ###
1. Go to **Recent items** or **Soon offline** listing
2. Select an episode and right-click, then choose **Go to program**
3. In the program episode listing, go back
4. Notice you end up in Kodi's Videos window

### Expected behavior ###
Kodi should go back to the originating **Recent items** or **Soon offline** listing.

### Additional context ###
- Operating system: LibreELEC v9.0.2
- Kodi version: v18.2
- Addon version: v2.2.0
- Using a VPN: no
- Country you are using the addon from: BE

**Comment by dagwieers** [comment]
This was reported upstream at: xbmc/xbmc#16492

**Comment by dagwieers** [comment]
We implemented a work-around using `XBMC.Container.Update(dir)`.

**Comment by dagwieers** [comment]
I am going to close this, as I have no hope this will be fixed in Kodi.

**Comment by dagwieers** [comment]
Thanks to @enen92 there is a fix for this in Kodi: https://github.com/xbmc/xbmc/pull/17893

**Comment by dagwieers** [comment]
The merged PR is for Matrix, I proposed it now for Leia as well.

**Comment by michaelarnauts** [comment]
> I am going to close this, as I have no hope this will be fixed in Kodi.

Never stop hoping! :)

**Comment by dagwieers** [comment]
This has now been merged into the Leia branch !

**Comment by dagwieers** [comment]
@mediaminister I cannot remember, is there a need to go back to using `ActivateWindow()`? Or do we keep `Container.Update()`.

**Comment by mediaminister** [comment]
No, there is no real need to go back to using `ActivateWindow()`, the current implementation works well enough.

In the past, there were timing issues updating resume points while Kodi Player was closing. But I implemented `wait_for_resumepoints` and this fixed it. So, most issues and comments regarding container updates before https://github.com/add-ons/plugin.video.vrt.nu/pull/652 are outdated.

FYI, you can find detailed information about the built-in functions at https://codedocs.xyz/AlwinEsch/kodi/page__list_of_built_in_functions.html
